ex5.2_cup_of_join.py

```diff
@@ -0,0 +1,9 @@
+#funcation that add the seperator for each list
+def print_join(*lists, seperator='-'):
+    list_with= [cell for curr_list in lists for cell in curr_list + [seperator]][:-1]
+    if list_with ==[]:
+        return 'empty'
```

Comment:
you should avoid returning different types from functions even though python allows it -  https://stackoverflow.com/questions/1839289/why-should-functions-always-return-the-same-type